packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/event_client.dart

```diff
@@ -215,21 +229,26 @@ class EventClient implements Closeable {
   }
 
   // If exception is recoverable, do not delete eventIds from local cache
-  void _handleRecoverableException(
-    SmithyHttpException e,
-    Map<String, StoredEvent> eventIdsToDelete,
-  ) {
+  void _logRecoverableException(Exception e) {
     _logger
       ..warn('putEvents - exception encountered: ', e)
       ..warn('Recoverable issue, will attempt to resend event batch');
-    eventIdsToDelete.clear();
   }
 
-  bool _isRetryable(String? message) {
-    return message != null &&
-        !_equalsIgnoreCase(message, 'ValidationException') &&
-        !_equalsIgnoreCase(message, 'SerializationException') &&
-        !_equalsIgnoreCase(message, 'BadRequestException');
+  bool _isRetryable(int statusCode) {
+    // Pinpoint service policy is for 500-599 status codes to be retryable
+    return statusCode >= 500 && statusCode < 600;
+  }
+
+  bool _shouldRetryEvent(String eventId) {
+    final timesFailed = _numFailuresByEvent[eventId] ?? 0;
+    if (timesFailed >= 3) {
```

Comment:
If we're in this method, we've failed at least once. If I understand this correctly, you're retrying 3 times, yeah? This is the logic needed for 3 retries:

```suggestion
    final timesFailed = _numFailuresByEvent[eventId] ?? 1;
    if (timesFailed >= 3) {
```